src/vs/workbench/browser/parts/titlebar/titlebarPart.ts

```diff
@@ -627,18 +627,43 @@ class ToogleConfigAction extends Action2 {
 	}
 }
 
-registerAction2(class ToogleCommandCenter extends ToogleConfigAction {
+registerAction2(class ToggleCommandCenter extends ToggleConfigAction {
 	constructor() {
 		super(LayoutSettings.COMMAND_CENTER, localize('toggle.commandCenter', 'Command Center'), 1);
 	}
 });
 
-registerAction2(class ToogleLayoutControl extends ToogleConfigAction {
+registerAction2(class ToggleLayoutControl extends ToggleConfigAction {
 	constructor() {
 		super('workbench.layoutControl.enabled', localize('toggle.layout', 'Layout Controls'), 2);
 	}
 });
 
+registerAction2(class ToggleEditorActions extends Action2 {
+	static readonly settingsID = `workbench.editor.editorActionsLocation`;
+	constructor() {
+		super({
+			id: `toggle.${ToggleEditorActions.settingsID}`,
+			title: localize('toggle.editorActions', 'Editor Actions'),
+			toggled: ContextKeyExpr.equals(`config.${ToggleEditorActions.settingsID}`, 'hidden').negate(),
+			menu: { id: MenuId.TitleBarContext, order: 3, when: ContextKeyExpr.equals(`config.workbench.editor.showTabs`, 'none') }
+		});
+	}
+
+	run(accessor: ServicesAccessor, ...args: any[]): void {
+		const configService = accessor.get(IConfigurationService);
+		const storageService = accessor.get(IStorageService);
+		const value = configService.getValue<string>(ToggleEditorActions.settingsID);
+		if (value === 'hidden') {
+			const storedValue = storageService.get(ToggleEditorActions.settingsID, StorageScope.PROFILE);
```

Comment:
Why does this involve a storage service?

Reply:
Because we need to store if the user had the setting 'default' or  'titleBar' when hiding the editor actions. This is the same as for the activityBar which also stores what setting you had before when hiding it.